File: src/lib/types/k8s/pods.ts
// Pod-related types
export interface PodPort {
  name?: string;
  containerPort: number;
  protocol: string;
}

export interface K8sPod {
  name: string;
  namespace: string;
  status: string;
  ready: string;
  restarts: number;
  age: string;
  ports?: PodPort[];
}

export interface PodFilters {
  labels?: string[];
  fieldSelectors?: string[];
}

File: src-tauri/src/k8s/types/pods.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct PodPort {
    pub name: Option<String>,
    pub container_port: i32,
    pub protocol: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct K8sPod {
    pub name: String,
    pub namespace: String,
    pub status: String,
    pub ready: String,
    pub restarts: i32,
    pub age: String,
    pub ports: Option<Vec<PodPort>>,
}
